File: Web/src/models/informationModels.ts
export interface InformationArticleListItem {
  id: string;
  title: string;
  thumbnail_image_url: string | null;
  published_at: string | null;
}

export interface InformationArticleResponse {
  id: string;
  title: string;
  content: string;
  thumbnail_image_url: string | null;
  view_count: number;
  is_published: boolean;
  published_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface InformationArticleCreateRequest {
  title: string;
  content: string;
  thumbnail_image_url?: string | null;
  is_published?: boolean;
}

export type InformationArticleUpdateRequest = Partial<InformationArticleCreateRequest>;

File: Android/app/src/main/java/com/diveapp/android/model/InformationModels.kt
package com.diveapp.android.model

import com.diveapp.android.core.network.OffsetDateTimeSerializer
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.time.OffsetDateTime

@Serializable
data class InformationArticleListItem(
    val id: String,
    val title: String,
    @SerialName("thumbnail_image_url") val thumbnailImageUrl: String?,
    @Serializable(with = OffsetDateTimeSerializer::class)
    @SerialName("published_at") val publishedAt: OffsetDateTime?,
)

@Serializable
data class InformationArticleResponse(
    val id: String,
    val title: String,
    val content: String,
    @SerialName("thumbnail_image_url") val thumbnailImageUrl: String?,
    @SerialName("view_count") val viewCount: Int,
    @SerialName("is_published") val isPublished: Boolean,
    @Serializable(with = OffsetDateTimeSerializer::class)
    @SerialName("published_at") val publishedAt: OffsetDateTime?,
)
